src/features/posts.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+export const init = createAsyncThunk('posts/fetch', (userId: number) => {
```

Comment:
```suggestion
export const getPosts = createAsyncThunk('posts/fetch', (userId: number) => {
```
use `getPosts` or `getPostsByUserId` etc. init is not clear name